lib/blocktree/blocktree.go

```diff
@@ -150,7 +153,19 @@ func (bt *BlockTree) GetAllBlocksAtDepth(hash common.Hash) []common.Hash {
 }
 
 // getNode finds and returns a node based on its Hash. Returns nil if not found.
-func (bt *BlockTree) getNode(h Hash) *node {
+func (bt *BlockTree) getNode(h Hash) (ret *node) {
+	defer func() {
+		if ret != nil {
+			if _, ok := bt.blockCache[ret.hash]; !ok {
+				bt.blockCache[ret.hash] = ret
```

Comment:
Would be nice to have a limit in the `blockCache` and when the `blockCache` reaches the limit then remove the first one index and insert the new `hash -> node`

Reply:
I'm not sure what criteria to use to remove the hashes... Would it be ok to remove randomly if the limit is reached??  Or maybe remove the least recently used adding a counter for each entry (I think that maybe this is a little overkill)??

What do you think?

Reply:
I don't think you need to do this, the blocktree shouldn't ever grow too large